Ghidra/Processors/NDS32/data/languages/nds32.sinc

```diff
@@ -0,0 +1,930 @@
+
+    ToggleL     = (21, 24)
+    Toggle      = (20, 20)
+
+    Usr         = (15, 19)
+    Group       = (10, 14)
+
+    DprefD      = (24, 24)
+    DprefSub    = (20, 23)
+
+    TlbopSub    = (5, 9)
+
+    StandbyZ    = (7, 9)
+    StandbySub  = (5, 6)
+    
+    GpSub1      = (19, 19)
+    GpSub2      = (18, 19)
+    GpSub3      = (17, 19)
+
+    LsmwRa = (15, 19)
+    LsmwRb = (20, 24)
+    LsmwRb_ = (20, 24)
+    LsmwRe = (10, 14)
+    LsmwRe_ = (10, 14)
+    Enable4 = (6, 9)
+    Enable4_fp = (9, 9)
+    Enable4_gp = (8, 8)
+    Enable4_lp = (7, 7)
+    Enable4_sp = (6, 6)
+    LsmwLs = (5, 5)
+    LsmwBa = (4, 4)
+    LsmwId = (3, 3)
+    LsmwM = (2, 2)
+    LsmwSub = (0, 1)
+;
+
+attach variables [Rt Rs Ra Rb Rd LsmwRa LsmwRb LsmwRe] [
+    a0 a1 a2 a3 a4 a5 s0 s1 s2 s3 s4 s5 s6 s7 s8 ta t0 t1 t2 t3 t4 t5 t6 t7 t8 t9 p0 p1 fp gp lp sp
+];
+
+attach variables [Rtl] [
+    a0 a2 a4 s0 s2 s4 s6 s8 t0 t2 t4 t6 t8 p0 fp lp
+];
+
+attach variables [Rth] [
+    a1 a3 a5 s1 s3 s5 s7 ta t1 t3 t5 t7 t9 p1 gp sp
+];
+
+attach variables [Dt] [
+    d0 d1
+];
+
+attach variables [Dtlow] [
+    d0.lo d1.lo
+];
+
+attach variables [Dthigh] [
+    d0.hi d1.hi
+];
+
+
+@define I32     "(OpSz=0)"
+@define ALU_1   "(Opc=0b100000)"
+@define ALU_2   "(Opc=0b100001)"
+@define ALU2Z   "(Alu2Mod=0b0000)"
+@define GPR     "(Alu2Mod=0b0001)"
+@define BR1     "(Opc=0b100110)"
+@define BR2     "(Opc=0b100111)"
+@define LSMW    "(Opc=0b011101)"
+@define JI      "(Opc=0b100100)"
+@define MEM     "(Opc=0b011100)"
+@define MISC    "(Opc=0b110010)"
+@define JREG    "(Opc=0b100101)"
+@define SBGP    "(Opc=0b011111)"
+@define LBGP    "(Opc=0b010111)"
+@define HWGP    "(Opc=0b011110)"
+@define SIMD    "(Opc=0b111000)"
+
+
+
+### ALU Instruction with Immediate ###
+
+:addi  Rt, Ra, Imm15s is $(I32) & Opc=0b101000 & Rt & Ra & Imm15s { Rt = Ra + Imm15s; }
+:subri Rt, Ra, Imm15s is $(I32) & Opc=0b101001 & Rt & Ra & Imm15s { Rt = Imm15s - Ra; }
+:andi  Rt, Ra, Imm15u is $(I32) & Opc=0b101010 & Rt & Ra & Imm15u { Rt = Ra & Imm15u; }
+:ori   Rt, Ra, Imm15u is $(I32) & Opc=0b101100 & Rt & Ra & Imm15u { Rt = Ra | Imm15u; }
+:xori  Rt, Ra, Imm15u is $(I32) & Opc=0b101011 & Rt & Ra & Imm15u { Rt = Ra ^ Imm15u; }
+:slti  Rt, Ra, Imm15s is $(I32) & Opc=0b101110 & Rt & Ra & Imm15s { Rt = zext(Ra < Imm15s); }
+:sltsi Rt, Ra, Imm15s is $(I32) & Opc=0b101111 & Rt & Ra & Imm15s { Rt = zext(Ra s< Imm15s); }
+:movi  Rt,     Imm20s is $(I32) & Opc=0b100010 & Rt & Imm20s      { Rt = Imm20s; }
+:sethi Rt,     Imm20u is $(I32) & Opc=0b100011 & Rt & Imm20u      { Rt = Imm20u << 12;}
+
+
+### ALU Instruction ###
+
+:add  Rt, Ra, Rb is $(I32) & $(ALU_1) & Rt & Ra & Rb & Rd=0 & Sub5=0b00000 { Rt = Ra + Rb; }
+:sub  Rt, Ra, Rb is $(I32) & $(ALU_1) & Rt & Ra & Rb & Rd=0 & Sub5=0b00001 { Rt = Ra - Rb; }
+:and  Rt, Ra, Rb is $(I32) & $(ALU_1) & Rt & Ra & Rb & Rd=0 & Sub5=0b00010 { Rt = Ra & Rb; }
+:xor  Rt, Ra, Rb is $(I32) & $(ALU_1) & Rt & Ra & Rb & Rd=0 & Sub5=0b00011 { Rt = Ra & Rb; }
```

Comment:
This should presumably be `Rt = Ra ^ Rb;`.

Reply:
Oops, I knew that I would eventually mess up my copy-pasting.
Nice catch, thanks for your comment.